src/plugins/discover/public/application/view_components/canvas/index.tsx

```diff
@@ -14,10 +14,18 @@ import { ResultStatus, SearchData } from '../utils/use_search';
 import { DiscoverNoResults } from '../../components/no_results/no_results';
 import { DiscoverUninitialized } from '../../components/uninitialized/uninitialized';
 import { LoadingSpinner } from '../../components/loading_spinner/loading_spinner';
+import { setColumns, useDispatch, useSelector } from '../../utils/state_management';
+import { filterColumns } from '../utils/filter_columns';
 
 // eslint-disable-next-line import/no-default-export
 export default function DiscoverCanvas({ setHeaderActionMenu, history }: ViewProps) {
   const { data$, refetch$, indexPattern } = useDiscoverContext();
+  const { columns } = useSelector((state) => state.discover);
+  const filteredColumns = filterColumns(columns, indexPattern);
+  const dispatch = useDispatch();
+
+  const prevColumns = useRef(filteredColumns);
```

Comment:
Doesn't look like we are using this. Do we need it?

Reply:
will remove.